### PR TITLE
[Localization] Updated and localized "The stack for this item is full." message

### PR DESCRIPTION
**Pull request by josericardo-fo**
## What are the changes?

Localized the "The stack for this item is full." message and changed it so it says which item has reached its stack limit.

## Why am I doing these changes?

The text was hardcoded and (thanks to @EnochG1) sometimes, e.g., at the start of [this daily run](https://www.youtube.com/watch?v=Ly4tODMs15w), you don't know which item has reached its stack limit.
## What did change?

Added the "itemStackFull" locale in battle.ts and changed the text to display the replaced item as well.
### Before (en)
image

### After (en)
image

### Before (pt_BR)
image

### After (pt_BR)
image

## How to test the changes?

Change these overrides:
```
readonly STARTING_WAVE_OVERRIDE: integer = 50;
readonly STARTING_LEVEL_OVERRIDE: integer = 100;
readonly STARTING_MODIFIER_OVERRIDE: Array<ModifierOverride> = [{ name: "GOLDEN_POKEBALL", count: 3 }];
```
Win the battle and check the message

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I considered writing automated tests for the issue?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/battle.ts [hunk 1]
```diff
@@ -57,6 +57,7 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "auswechseln",
   "escapeVerbFlee": "flucht",
   "skipItemQuestion": "Bist du sicher, dass du kein Item nehmen willst?",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:60]:
```suggestion
"itemStackFull": "Du hast bereits zu viele von diesem Gegenstand. Du erhältst stattdessen {{itemName}}.",
```

File: src/locales/fr/battle.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}} récupère des PV !",
   "hpIsFull": "Les PV de {{pokemonName}}\nsont au maximum !",
   "skipItemQuestion": "Êtes-vous sûr·e de ne pas vouloir prendre d’objet ?",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:63]:
```suggestion
  "itemStackFull": "Vous avez atteint la quantité maximale pour cet objet.\nVous recevez {{itemName}} à la place.",
```

File: src/locales/es/battle.ts [hunk 3]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
   "hpIsFull": "{{pokemonName}}'s\nHP is full!",
   "skipItemQuestion": "¿Estás seguro de que no quieres coger un objeto?",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:63]:
```suggestion
  "itemStackFull": "El máximo de este objeto ha sido alcanzado.\nRecibirás {{itemName}} en su lugar.",
```

File: src/locales/es/battle.ts [hunk 4]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
   "hpIsFull": "{{pokemonName}}'s\nHP is full!",
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:61]:
```suggestion
  "turnEndHpRestore": "Los PS de {{pokemonName}} fueron restaurados.",
  "hpIsFull": "¡Los PS de {{pokemonName}}\nestán al máximo!",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added es translation and fixed other 2 es lines

File: src/locales/de/battle.ts (hunk 1 above)
Comment (josericardo-fo) [reply src/locales/de/battle.ts:60]: @CodeTappert you removed blank spaces at the beggining

File: src/locales/ko/battle.ts [hunk 5]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}의\n체력이 회복되었다!",
   "hpIsFull": "그러나 {{pokemonName}}의\n체력이 가득 찬 상태다!",
   "skipItemQuestion": "아이템을 받지 않고 넘어가시겠습니까?",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (EnochG1) [inline src/locales/ko/battle.ts:63]:
```suggestion
  "itemStackFull": "이 아이템의 소지 한도에 도달했습니다.\n{{itemname}}[[를]] 대신 받습니다.",
```

Comment (sodaMelon) [reply src/locales/ko/battle.ts:63]: notice: There's a semantic issue, so we're rethinking it.

Comment (EnochG1) [reply src/locales/ko/battle.ts:63]: @sodaMelon I would change this if below suggestion is approved.

File: src/locales/zh_CN/battle.ts [hunk 6]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}的体力恢复了。",
   "hpIsFull": "{{pokemonName}}的体力已满！",
   "skipItemQuestion": "你确定要跳过拾取道具吗？",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:63]:
```suggestion
  "itemStackFull": "该道具持有数达到上限，\n你获得了{{itemName}}作为替代。",
```

**Review by 40chyan — COMMENTED** [review]
zh_cn

File: src/locales/zh_TW/battle.ts [hunk 7]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
   "hpIsFull": "{{pokemonName}}'s\nHP is full!",
   "skipItemQuestion": "你要跳過拾取道具嗎？",
+  "itemStackFull": "The stack for this item is full.\nYou will receive {{itemName}} instead.",
```
Comment (mercurius-00) [inline src/locales/zh_TW/battle.ts:63]:
```suggestion
  "itemStackFull": "該道具持有數已達到上限，\n你獲得了{{itemName}}作爲代替。",
```

**Review by mercurius-00 — COMMENTED** [review]
zh-tw

File: src/locales/ko/battle.ts [hunk 8]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}의\n체력이 회복되었다!",
   "hpIsFull": "그러나 {{pokemonName}}의\n체력이 가득 찬 상태다!",
   "skipItemQuestion": "아이템을 받지 않고 넘어가시겠습니까?",
+  "itemStackFull": "이 아이템의 소지 한도에 도달했습니다.\n{{itemname}}[[를]] 대신 받습니다.",
```
Comment (EnochG1) [inline src/locales/ko/battle.ts:63]:
```suggestion
  "itemStackFull": "{{fullItemName}}의 소지 한도에 도달했습니다.\n{{itemname}}[[를]] 대신 받습니다.",
```

Comment (josericardo-fo) [reply src/locales/ko/battle.ts]: @EnochG1 just edit the suggestion again plz

File: src/locales/fr/battle.ts [hunk 9]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}} récupère des PV !",
   "hpIsFull": "Les PV de {{pokemonName}}\nsont au maximum !",
   "skipItemQuestion": "Êtes-vous sûr·e de ne pas vouloir prendre d’objet ?",
+  "itemStackFull": "Vous avez atteint la quantité maximale pour cet objet.\nVous recevez {{itemName}} à la place.", // Check the english translation
```
Comment (Adri1) [inline src/locales/fr/battle.ts:63]:
```suggestion
  "itemStackFull": "Quantité maximale de {{fullItemName}} atteinte.\nVous recevez {{itemName}} à la place.",
```

File: src/locales/de/battle.ts [hunk 10]
```diff
@@ -57,6 +57,7 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "auswechseln",
   "escapeVerbFlee": "flucht",
   "skipItemQuestion": "Bist du sicher, dass du kein Item nehmen willst?",
+  "itemStackFull": "Du hast bereits zu viele von diesem Gegenstand. Du erhältst stattdessen {{itemName}}.", // Check the english translation
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:60]:
```suggestion
  "itemStackFull": "Du hast bereits zu viele von {{fullItemName}}. Du erhältst stattdessen {{itemName}}.",
```

File: src/locales/es/battle.ts [hunk 11]
```diff
@@ -57,9 +57,10 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "cambiar",
   "escapeVerbFlee": "huir",
   "notDisabled": "¡El movimiento {{moveName}} de {{pokemonName}}\nya no está anulado!",
-  "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
-  "hpIsFull": "{{pokemonName}}'s\nHP is full!",
+  "turnEndHpRestore": "Los PS de {{pokemonName}} fueron restaurados.",
+  "hpIsFull": "¡Los PS de {{pokemonName}}\nestán al máximo!",
   "skipItemQuestion": "¿Estás seguro de que no quieres coger un objeto?",
+  "itemStackFull": "El máximo de este objeto ha sido alcanzado.\nRecibirás {{itemName}} en su lugar.", // Check the english translation
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:63]:
```suggestion
  "itemStackFull": "El máximo número de {{fullItemName}} ha sido alcanzado.\nRecibirás {{itemName}} en su lugar.",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added ES

**Review by protimita — COMMENTED** [review]
JA translation

**Comment by josericardo-fo** [comment]
> JA translation

where?

File: src/locales/es/battle.ts [hunk 12]
```diff
@@ -57,9 +57,10 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "cambiar",
   "escapeVerbFlee": "huir",
   "notDisabled": "¡El movimiento {{moveName}} de {{pokemonName}}\nya no está anulado!",
-  "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
-  "hpIsFull": "{{pokemonName}}'s\nHP is full!",
+  "turnEndHpRestore": "Los PS de {{pokemonName}} fueron restaurados.",
+  "hpIsFull": "¡Los PS de {{pokemonName}}\nestán al máximo!",
   "skipItemQuestion": "¿Estás seguro de que no quieres coger un objeto?",
+  "itemStackFull": "El máximo número de {{fullItemName}} ha sido alcanzado.\nRecibirás {{itemName}} en su lugar.",
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:63]:
```suggestion
  "itemStackFull": "El máximo número de {{fullItemName}} ha sido alcanzado. Recibirás {{itemName}} en su lugar.",
```

**Review by AsdarDevelops — COMMENTED** [review]
Removed the "\n" out of fear text might be too long.

**Review by CodeTappert — APPROVED** [review]

File: src/locales/ja/battle.ts [hunk 13]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}の　体力が　回復した！",
   "hpIsFull": "{{pokemonName}}の\n体力が　満タンだ!",
   "skipItemQuestion": "本当に　アイテムを　取らずに　進みますか？",
+  "itemStackFull": "The stack for {{fullItemName}} is full.\nYou will receive {{itemName}} instead.",
```
Comment (protimita) [inline src/locales/ja/battle.ts:63]:
```suggestion
  "itemStackFull": "{{fullItemName}}の　スタックが　いっぱいです。\n代わりに　{{itemName}}を　取得します。",
```

**Review by Adri1 — APPROVED** [review]

File: src/locales/zh_TW/battle.ts [hunk 14]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
   "hpIsFull": "{{pokemonName}}'s\nHP is full!",
   "skipItemQuestion": "你要跳過拾取道具嗎？",
+  "itemStackFull": "該道具持有數已達到上限，\n你獲得了{{itemName}}作爲代替。", // Check the english translation
```
Comment (mercurius-00) [inline src/locales/zh_TW/battle.ts:63]:
```suggestion
  "itemStackFull": "{{fullItemName}}持有數已達到上限，\n你獲得了{{itemName}}作爲代替。", 
```

File: src/locales/zh_CN/battle.ts [hunk 15]
```diff
@@ -60,6 +60,7 @@ export const battle: SimpleTranslationEntries = {
   "turnEndHpRestore": "{{pokemonName}}的体力恢复了。",
   "hpIsFull": "{{pokemonName}}的体力已满！",
   "skipItemQuestion": "你确定要跳过拾取道具吗？",
+  "itemStackFull": "该道具持有数达到上限，\n你获得了{{itemName}}作为替代。", // Check the english translation
```
Comment (mercurius-00) [inline src/locales/zh_CN/battle.ts:63]:
```suggestion
  "itemStackFull": "{{fullItemName}}持有数达到上限，\n你获得了{{itemName}}作为替代。", // Check the english translation
```

**Review by mercurius-00 — COMMENTED** [review]
zh

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**